pynicotine/gtkgui/widgets/textentry.py

```diff
@@ -147,13 +147,8 @@ def set_entry(self, entry):
 
     def add_completion(self, item):
 
-        if not config.sections["words"]["tab"]:
-            return
-
-        if item in self.completion_list:
-            return
-
-        self.completion_list.append(item)
+        if config.sections["words"]["tab"] and item not in self.completion_list:
```

Comment:
The completions are always supposed to be updated when tab/popup completion is enabled/disabled (see https://github.com/nicotine-plus/nicotine-plus/blob/4d402687ecb7802668ea1eca522035540b74388c/pynicotine/gtkgui/dialogs/preferences.py#L2760-L2762), so `item not in self.completion_list` should ideally not be required.

Reply:
Is that not also needed to avoid duplicates? Such as if a user is both in a chat room and is a buddy, for example.

Reply:
Duplicates are removed earlier: https://github.com/nicotine-plus/nicotine-plus/blob/4d402687ecb7802668ea1eca522035540b74388c/pynicotine/gtkgui/chatrooms.py#L1228-L1232

Reply:
Sorry, I forgot that `add_completion()` has a check as well. That one should indeed remain in place.

Reply:
But there is another instance in user_joined_room() where items get added without checking for duplicates: 

https://github.com/nicotine-plus/nicotine-plus/blob/4d402687ecb7802668ea1eca522035540b74388c/pynicotine/gtkgui/chatrooms.py#L950-L951

Reply:
The duplicate checking could be avoided if `completion_list = []` was instead `completion_words = set()`